### DISCUSSION TITLE
Why does `poetry lock --no-update` produce different `poetry.lock` output in a Linux container vs MacOS

### DISCUSSION THREAD

**Discussion by allanlewis** [python-poetry/poetry#8934] [Q&A] [answered]
I use a Mac (Intel chip) and my development environment uses Linux Docker containers based on the official Python images. If I run `poetry lock --no-update` with one of my projects, the lock-file differs slightly depending on whether I do that in the Docker container vs locally. I'm using the same version of Poetry in each case.

Specifically, it seems that wheels for other platforms are removed, e.g. I have `psycopg2-binary` and `PyYAML` in my dependencies and the following files are removed when I run Poetry locally (on Mac) compared to a baseline run in the (Linux) container:

```
psycopg2_binary-2.9.9-cp312-cp312-macosx_11_0_arm64.whl
psycopg2_binary-2.9.9-cp312-cp312-win32.whl
psycopg2_binary-2.9.9-cp312-cp312-win_amd64.whl
PyYAML-6.0.1-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl
```

The common factor between these files is that they are either for platforms other than Mac or Linux, or they're for ARM CPUs.

Is this expected behaviour or have I discovered a bug? I didn't want to raise an issue in case this is as designed, but it seems odd to me.

**Comment by Secrus** [comment, marked as answer]
Most likely your MacOS cache is old. Try running `poetry cache clear PyPI --all` and see if you can reproduce the issue.

**Reply by allanlewis** [reply]
Thanks, that solved it!
